src/VisualStudio/Core/Def/Shared/VisualStudioImageIdService.cs

```diff
@@ -47,32 +49,32 @@ internal class VisualStudioImageMonikerService : ForegroundThreadAffinitizedObje
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public VisualStudioImageMonikerService(IThreadingContext threadingContext, SVsServiceProvider serviceProvider)
+        public VisualStudioImageIdService(IThreadingContext threadingContext, SVsServiceProvider serviceProvider)
             : base(threadingContext)
         {
             _imageService = (IVsImageService2)serviceProvider.GetService(typeof(SVsImageService));
         }
 
-        public bool TryGetImageMoniker(ImmutableArray<string> tags, out ImageMoniker imageMoniker)
+        public bool TryGetImageId(ImmutableArray<string> tags, out ImageId imageId)
         {
             this.AssertIsForeground();
 
-            imageMoniker = GetImageMoniker(tags);
-            return !imageMoniker.IsNullImage();
+            imageId = GetImageId(tags);
+            return imageId != default;
         }
 
-        private ImageMoniker GetImageMoniker(ImmutableArray<string> tags)
+        private ImageId GetImageId(ImmutableArray<string> tags)
         {
             var glyph = tags.GetFirstGlyph();
             switch (glyph)
             {
                 case Glyph.AddReference:
-                    return GetCompositedImageMoniker(
+                    return GetCompositedImageId(
```

Comment:
I have no idea if this code is ever hit in reality, because it's ugly, but for the record it does still work:
![image](https://user-images.githubusercontent.com/754264/114489544-aab45e80-9c56-11eb-81a1-84842d72ba88.png)

Reply:
And just to confirm that this change didn't break anything, its just as ugly in main:
![image](https://user-images.githubusercontent.com/754264/114490583-75107500-9c58-11eb-97a3-b8bc5c86511c.png)